RecoHGCal/TICL/plugins/PFTICLProducer.cc

```diff
@@ -59,6 +61,7 @@ void PFTICLProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptio
   desc.add<edm::InputTag>("trackTimeErrorMap", edm::InputTag("tofPID:sigmat0"));
   desc.add<edm::InputTag>("trackTimeQualityMap", edm::InputTag("mtdTrackQualityMVA:mtdQualMVA"));
   desc.add<double>("timingQualityThreshold", 0.5);
+  desc.addOptional<bool>("useMTDTiming", true);
```

Comment:
Why `addOptional`? The parameter is always used in the code, see L149 below
```suggestion
  desc.add<bool>("useMTDTiming", true);
```

In https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp#Optional_Parameters it is written " During validation if an optional parameter is missing, then it is just ignored. Nothing is inserted and nothing is thrown. " Better to insert the `true` default value, instead

Reply:
@perrotta My idea was that even old configurations would continue running with no change... i.e. to be the least intrusive possible. Do you think it is better to not have it optional?

Reply:
> @perrotta My idea was that even old configurations would continue running with no change... i.e. to be the least intrusive possible. Do you think it is better to not have it optional?

This is what a default value is needed for.
Having it as "optional" means that there are cases in which that parameter is not needed in the workflow of the program, i.e. it can remain indefinite. This is definitely not the case, because at L149 you do use it.